src/AppInstallerCLIE2ETests/InstallCommand.cs

```diff
@@ -155,6 +163,79 @@ public void InstallExeWithAlternateSourceFailure()
+            TestCommon.VerifyPortablePackage(Path.Combine(installDir, packageDirName), commandAlias, fileName, productCode, true);
+        }
+
+        [Test]
+        public void InstallPortableExeWithCommand()
+        {
+            var installDir = TestCommon.GetRandomTestDir();
+            string packageId, commandAlias, fileName, productCode;
+            packageId = "AppInstallerTest.TestPortableExeWithCommand";
+             productCode = packageId + "_" + TestSourceIdentifier;
+            fileName = "AppInstallerTestExeInstaller.exe";
+            commandAlias = "testCommand.exe";
+
+            var result = TestCommon.RunAICLICommand("install", $"{packageId} -l {installDir}");
+            Assert.AreEqual(Constants.ErrorCode.S_OK, result.ExitCode);
+            Assert.True(result.StdOut.Contains("Successfully installed"));
+            TestCommon.VerifyPortablePackage(installDir, commandAlias, fileName, productCode, true);
+        }
+
+        [Test]
+        public void InstallPortableExeWithRename()
+        {
+            var installDir = TestCommon.GetRandomTestDir();
+            string packageId, productCode, renameArgValue;
+            packageId = "AppInstallerTest.TestPortableExeWithCommand";
+            productCode = packageId + "_" + TestSourceIdentifier;
+            renameArgValue = "testRename.exe";
+
+            var result = TestCommon.RunAICLICommand("install", $"{packageId} -l {installDir} --rename {renameArgValue}");
+            Assert.AreEqual(Constants.ErrorCode.S_OK, result.ExitCode);
+            Assert.True(result.StdOut.Contains("Successfully installed"));
+            TestCommon.VerifyPortablePackage(installDir, renameArgValue, renameArgValue, productCode, true);
+        }
+
+        [Test]
+        public void InstallPortableInvalidRename()
+        {
+            var installDir = TestCommon.GetRandomTestDir();
+            string packageId, renameArgValue;
+            packageId = "AppInstallerTest.TestPortableExeWithCommand";
+            renameArgValue = "test?";
+
+            var result = TestCommon.RunAICLICommand("install", $"{packageId} -l {installDir} --rename {renameArgValue}");
+            Assert.AreNotEqual(Constants.ErrorCode.S_OK, result.ExitCode);
+            Assert.True(result.StdOut.Contains("The specified filename is not a valid filename"));
+        }
+
+        [Test]
+        public void InstallPortableReservedNames()
+        {
+            var installDir = TestCommon.GetRandomTestDir();
+            string packageId, renameArgValue;
+            packageId = "AppInstallerTest.TestPortableExeWithCommand";
+            renameArgValue = "CON";
+
+            var result = TestCommon.RunAICLICommand("install", $"{packageId} -l {installDir} --rename {renameArgValue}");
+            Assert.AreNotEqual(Constants.ErrorCode.S_OK, result.ExitCode);
+            Assert.True(result.StdOut.Contains("The parameter is incorrect."));

```

Comment:
If this is the expected result, it doesn't seem like it would be very clear to the user that a reserved name is not a valid file name. As a user, I would expect that the output of a Reserved name is similar to that of an InvalidRename

Reply:
Fixed handling of reserved words to produce the proper error message.